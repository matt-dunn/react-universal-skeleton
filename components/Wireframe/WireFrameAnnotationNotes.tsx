import React from "react";
import styled from "@emotion/styled";
import {IdentifierBase} from "./styles";
import {WireFrameComponent, WireFrameComponents} from "./api";

type WireFrameAnnotationsNotesProps = {
    components: WireFrameComponents;
    highlightedNote?: WireFrameComponent;
}

const WireFrameAnnotationsNotesContainer = styled.ul`
  overflow: auto;
  font-size: 0.85em;
  
  li {
    padding: 6px 0;
    border-bottom: 1px solid #ccc;
    
    &:last-child {
      border-bottom: none;
    }
    
    &.highlight {
      background-color: rgba(64, 134, 247, 0.25);
    }
    
    header {
      display: flex;
      padding: 0 10px;
      margin-bottom: 5px;
      align-items: center;
    }
    
    article {
      padding: 0 10px;
    }
  }
`;

const IdentifierNote = styled(IdentifierBase)`
  margin-right: 0.5em;
`;

export const WireFrameAnnotationsNotes = ({components, highlightedNote}: WireFrameAnnotationsNotesProps) => {
    return (
        <WireFrameAnnotationsNotesContainer id="wf-annotations">
            {components.map(component => (
                <li
                    key={component.id}
                    id={`wf-annotation-${component.id}`}
                    className={(highlightedNote === component && "highlight") || ""}
                >
                    <header>
                        <IdentifierNote>
                            {component.id}
                        </IdentifierNote>
                        <h2>
                            {component.options.title}
                        </h2>
                    </header>
                    <article>
                        {component.options.description}
                    </article>
                </li>
            ))}
        </WireFrameAnnotationsNotesContainer>
    );
};
