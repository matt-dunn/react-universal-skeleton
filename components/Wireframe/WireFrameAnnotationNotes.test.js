import React from "react";
import {shallow} from "enzyme";

import {WireFrameAnnotationsNotes as Component} from "./WireFrameAnnotationNotes";

describe("Wireframe: WireFrameAnnotationsNotes", () => {
  let MockedComponent,
    components;

  beforeEach(() => {
    MockedComponent = jest.fn();

    components = [
      {
        id: 1,
        count: 1,
        Component: MockedComponent,
        options: {
          title: "Test component",
          description: "Test description"
        }
      }
    ];
  });

  it("renders correctly", () => {
    const highlightedNote = undefined;

    const wrapper = shallow(
      <Component
        components={components}
        highlightedNote={highlightedNote}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with highlight", () => {
    const highlightedNote = components[0];

    const wrapper = shallow(
      <Component
        components={components}
        highlightedNote={highlightedNote}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
