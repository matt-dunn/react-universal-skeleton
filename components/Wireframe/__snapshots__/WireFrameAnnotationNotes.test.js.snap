// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wireframe: WireFrameAnnotationsNotes renders correctly 1`] = `
<WireFrameAnnotationsNotesContainer
  id="wf-annotations"
>
  <li
    className=""
    id="wf-annotation-1"
    key="1"
  >
    <header>
      <IdentifierNote>
        1
      </IdentifierNote>
      <h2>
        Test component
      </h2>
    </header>
    <article>
      Test description
    </article>
  </li>
</WireFrameAnnotationsNotesContainer>
`;

exports[`Wireframe: WireFrameAnnotationsNotes renders correctly with highlight 1`] = `
<WireFrameAnnotationsNotesContainer
  id="wf-annotations"
>
  <li
    className="highlight"
    id="wf-annotation-1"
    key="1"
  >
    <header>
      <IdentifierNote>
        1
      </IdentifierNote>
      <h2>
        Test component
      </h2>
    </header>
    <article>
      Test description
    </article>
  </li>
</WireFrameAnnotationsNotesContainer>
`;
